hpx/parallel/segmented_algorithms/detail/dispatch.hpp

```diff
@@ -145,6 +178,49 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail
         }
     };
 
+    template <typename Iterator1, typename Iterator2, typename Iterator3>
+    struct algorithm_result_helper<
+        future<hpx::util::tuple<Iterator1, Iterator2, Iterator3> >,
+        typename std::enable_if<
+                hpx::traits::is_segmented_local_iterator<Iterator1>::value ||
+                hpx::traits::is_segmented_local_iterator<Iterator2>::value ||
+                hpx::traits::is_segmented_local_iterator<Iterator3>::value
+            >::type>
+    {
+        typedef hpx::traits::segmented_local_iterator_traits<Iterator1> traits1;
+        typedef hpx::traits::segmented_local_iterator_traits<Iterator2> traits2;
+        typedef hpx::traits::segmented_local_iterator_traits<Iterator3> traits3;
+
+        typedef hpx::util::tuple<
+                typename traits1::local_raw_iterator,
+                typename traits2::local_raw_iterator,
+                typename traits3::local_raw_iterator
+            > arg_type;
+
+        static HPX_FORCEINLINE
+        future<hpx::util::tuple<
+            typename traits1::local_iterator, typename traits2::local_iterator,
+            typename traits3::local_iterator
+        > >
+        call(future<arg_type> && f)
+        {
+            return f.then(
+                [](future<arg_type> && f)
+                ->  hpx::util::tuple<
+                        typename traits1::local_iterator,
+                        typename traits2::local_iterator,
+                        typename traits3::local_iterator
+                    >
+                {
+                    auto p = f.get();
```

Comment:
I reverted this change while trying to figure out the errors with transform. I will make this change (auto p to auto && p) on this and other helpers again.